ui update for signup, login and chat screen

akinolu52 opened the pull request:
fix for [PR #2](https://github.com/akinolu52/turbo-rotary/issues/2)

<img width="1440" alt="Screenshot 2021-08-06 at 21 55 39" src="https://user-images.githubusercontent.com/16763860/128570392-b64b1796-ce6a-4a4d-b896-68575a4d5575.png">
<img width="1440" alt="Screenshot 2021-08-06 at 21 55 46" src="https://user-images.githubusercontent.com/16763860/128570420-0ee50906-34a4-4a4d-a5f3-54b1776bec12.png">
<img width="1440" alt="Screenshot 2021-08-06 at 21 56 08" src="https://user-images.githubusercontent.com/16763860/128570421-9355d71b-1f7e-4c63-9df1-ba301dab1b2d.png">

akinolu52 committed:
ui update for signup, login and chat screen

FilipeGorodscy commented on client/src/Login.js:
```diff
@@ -8,12 +8,68 @@ import {
+    position: 'relative',
+    display: 'flex',
+    // flexDirection: "column",
+    padding: '2rem 6rem',
```
You can remove commented-out code from your PRs

akinolu52 replied:
okay

FilipeGorodscy commented on client/src/Login.js:
```diff
@@ -23,55 +79,67 @@ const Login = (props) => {
+                />
+              </FormControl>
+            </Grid>
+            <div style={{ width: '90%', textAlign: 'center' }}>
```
You should not mix JSX with Material UI and you should avoid inline styles.
All `<div>` occurrences should be changed for `<Box>` or `<Grid>` (depending on what properties you might want)

FilipeGorodscy commented on client/src/Login.js:
```diff
@@ -23,55 +79,67 @@ const Login = (props) => {
+                  InputProps={{
+                    endAdornment: (
+                      <InputAdornment position='end' className={classes.labelFocusStyle}>
+                        <span>Forgot Password?</span>
```
`<span>` should be changed to `<Typography>`

FilipeGorodscy commented on client/src/components/Sidebar/CurrentUser.js:
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
+  };
+
+  const handleLogout = async () => {
+    console.log('yes')
```
You can remove console logs from your PRs

FilipeGorodscy commented on client/src/components/Sidebar/CurrentUser.js:
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
+  const [open, setOpen] = useState(false);
+  let { logoutUser, user } = props;
+
+  user = user || {};
```
```suggestion
  const { logoutUser, user={} } = props;
```

akinolu52 replied:
okay, thanks

FilipeGorodscy commented on client/src/components/Sidebar/CurrentUser.js:
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
+                  Logout
+                </Button>
+              </div>
+            ) : null}
```
```suggestion
            {open && (
              <div className={classes.dropdown}>
                <Button className={classes.button1}
                  onClick={handleLogout}
                >
                  Logout
                </Button>
              </div>
            )}
```

FilipeGorodscy reviewed (commented):
Good job overall. Some notes:
- It looks like some of your font colours (especially the phrase on the top right corner) are a little off compared to the mock design.
- There are a lot of repeated styles/components between Login/Signup pages.
- It looks like the linear gradient in the background image for Login/Signup pages is different from the mock design.

I'll attach a screenshot of the mobile view of the application so the other reviewers can reference it. Please let us know on Slack when this PR is ready for a second review!
<img width="496" alt="Screen Shot 2021-08-06 at 5 24 58 PM" src="https://user-images.githubusercontent.com/53091098/128573635-2b29da52-e159-4c0f-9da1-9f5d70ffefd9.png">

akinolu52 committed:
1. fixed mobile device views
2. new auth layout

akinolu52 commented:
PR updated

<img width="619" alt="Screenshot 2021-08-07 at 16 16 36" src="https://user-images.githubusercontent.com/16763860/128605032-0c82259d-b805-48cc-b181-ea843c77d029.png">

akinolu52 committed:
fixed linear gradient image

cassandrale179 commented on client/src/Login.js:
```diff
@@ -1,19 +1,40 @@
+  },
+  button: {
+    color: '#FFF',
+    backgroundColor: "#3A8DFF",
```
Nit: we can use theme here for color as well

akinolu52 replied:
okay

cassandrale179 commented on client/src/Signup.js:
```diff
@@ -1,118 +1,92 @@
+  },
+  button: {
+    color: '#FFF',
+    backgroundColor: "#3A8DFF",
```
Same here

akinolu52 replied:
thanks, done

cassandrale179 commented on client/src/components/Layout/AuthLayout.js:
```diff
@@ -0,0 +1,123 @@
+            flexDirection: 'column'
+        },
+    },
+    left: {
```
Nit: left is not a descriptive name, let's change this to .SideBar or something similar.

cassandrale179 commented on client/src/components/Layout/AuthLayout.js:
```diff
@@ -0,0 +1,123 @@
+            marginBottom: '1rem'
+        },
+    },
+    right: {
```
Nit, same here, let's rename this class to be something more descriptive other than 'right'

cassandrale179 commented on client/src/components/Layout/AuthLayout.js:
```diff
@@ -0,0 +1,123 @@
+            padding: '1.2rem 1.5rem'
+        },
+    },
+    button1: {
```
Same here, let's rename this from button1 and button2 to what the button represent, such as 'FormButton' or 'HeaderButton'

akinolu52 replied:
thanks

cassandrale179 commented on client/src/components/Sidebar/CurrentUser.js:
```diff
@@ -29,20 +31,63 @@ const useStyles = makeStyles(() => ({
-        <MoreHorizIcon classes={{ root: classes.ellipsis }} />
+
+        <ClickAwayListener onClickAway={handleClickAway}>
+          <div className={classes.dropdownRoot}>
```
To keep it consistent with the code base, let's not mix regular JSX expression with Material UI and used Grid or Box when appropriate.

akinolu52 committed:
1. removed unused styles
2. renamed sidebar and button classes
3. used theme to control the button color
